### PR TITLE
Scenario test with anoncreds wallet upgrade and restart

### PR DESCRIPTION
**Pull request by ianco**
Also a bit of refactoring ...

Alice (issuer - askar) issues 2 credentials and revokes 1 to each of 3 holders (2 askar and 1 anoncreds).

The revocation registry size is set to `5` to ensure there will be a `full` registry.

Alice and one of the askar holders are then upgraded to anoncreds wallets.

### CODE REVIEW

**Comment by jamshale** [comment]
Looks good 👍 Happy to have this scenario tested better.

File: scenarios/examples/util.py [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+    inspect_results = container.attrs
+    return not inspect_results["State"]["Running"]
+
+
+def wait_until_healthy(client, container_id: str, attempts: int = 350, is_healthy=True):
+    """Wait until container is healthy."""
+    container = client.containers.get(container_id)
+    print((container.name, container.status))
+    for _ in range(attempts):
+        if (is_healthy and healthy(container)) or unhealthy(container):
```
Comment (swcurran) [inline scenarios/examples/util.py:40]: I gotta ask... "healthy" or "unhealthy" is fine??

Comment (ianco) [reply scenarios/examples/util.py:40]: "unhealthy" means we're shutting down the agent and we're checking that it's actually down (we let it shutdown cleanly).

Comment (swcurran) [reply scenarios/examples/util.py:40]: Sounds good.

**Review by jamshale — APPROVED** [review]
Looks really good 👍This is what I was hoping for. Such a useful testing tool to have.

Only thing I'm not sure about is if the credentials actual revocation status is being tested after the upgrade. I remember being a bit concerned that the indexing would line up and the revocation size was identical. I can't remember exactly. There was an issue with this at one point that got fixed.

**Comment by ianco** [comment]
> Only thing I'm not sure about is if the credentials actual revocation status is being tested after the upgrade.

Good point, I'll add some more post-upgrade testing to confirm everything is upgraded properly (for both issuer and holder).

**Pull request merged**